pkg/apis/operator/v1alpha1/types.go

```diff
@@ -445,6 +449,19 @@ type GardenerAPIServerConfig struct {
 	WatchCacheSizes *gardencorev1beta1.WatchCacheSizes `json:"watchCacheSizes,omitempty"`
 }
 
+// GardenerAdmissionControllerConfig contains configuration settings for the gardener-admission-controller.
+type GardenerAdmissionControllerConfig struct {
+	// LogLevel is the configured log level for the gardener-admission-controller. Must be one of [info,debug,error].
+	// Defaults to info.
+	// +kubebuilder:validation:Enum=info;debug;error
+	// +kubebuilder:default=info
+	// +optional
+	LogLevel *string `json:"logLevel,omitempty"`
+	// ResourceAdmissionConfiguration is the configuration for resource size restrictions for arbitrary Group-Version-Kinds.
+	// +optional
+	ResourceAdmissionConfiguration *admissioncontrollerv1alpha1.ResourceAdmissionConfiguration `json:"resourceAdmissionConfiguration,omitempty"`
```

Comment:
In `GardenerControllerManager`, I had a similar problem with using GCMs' `ProjectQuotaConfiguration` type. In the end, I decided not to reuse it from GCM's `config` package but "duplicate" it here in the `Garden` API.

Should we do the same here for GAC to avoid pulling in "ugly" transitive dependencies? I see you already modified some import restrictions to make the linter happy, but logically we probably rather don't want this? 👀 

OTOH, for GAC it's not "too bad", we just pull the API and validation packages. For GCM it was much worse. Then again, it's inconsistent now in the `Garden` API 🤷🏻

Reply:
I was also torn between the two options, reuse vs. copy. The `ResourceAdmissionConfiguration` struct is more complex as `ProjectQuotaConfiguration` and would involve more copied code, esp. the validation.

> OTOH, for GAC it's not "too bad"

This was my final thought and decision to reuse the struct. I don't think that it makes the API inconsistent because it's already inconsistent in that sense today: The `Garden` API reuses many structs from the `gardencorev1beta1` API group. Yes, you can argue that this is no controller config API but still 😄

Reply:
> This was my final thought and decision to reuse the struct. I don't think that it makes the API inconsistent because it's already inconsistent in that sense today: The `Garden` API reuses many structs from the `gardencorev1beta1` API group. Yes, you can argue that this is no controller config API but still 😄

Yes, that's indeed the point - the `core.gardener.cloud/v1beta1` API is end-user facing, beta ("stable"), and changes follow proper deprecation periods. Hence, it's fine to important it in other such APIs like `operator.gardener.cloud/v1alpha1`. The component config APIs are quasi-internal, alpha, and changes are typically breaking w/o proper process. When imported, such "aggressive changes" would also affect the "stable" APIs.

We are already in such situation with the `ManagedSeed` and the gardenlet config, although we use `runtime.RawExtension` there which is no option in this PR/scope.

I see two options:
1. Copy the structs of the GAC config for the `Garden` API, and translate them to call the GAC validation functions without needing to duplicating them as well.
   - This way, we cannot accidentally break the `Garden` API when changing the component config API of GAC.
2. Keep it is as and reuse the GAC config also in the `Garden` API
   - But then we should probably find a way(/process) to make sure we don't accidentally break it (how?)
   - This also has the downside that we have to add the workarounds in the import restrictions to make the linter happy even though we don't really need/want to allow such imports (e.g., in `/extensions/.import-restrictions`)

Reply:
I followed the first suggestion in [f822459](https://github.com/gardener/gardener/pull/8251/commits/f8224593a99ba61f5915c3ff45a157bc0f81f055). PTAL.